tests/src/org.apache.curator/curator-framework/5.5.0/src/test/java/org_apache_curator/curator_framework/CuratorFrameworkTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_curator.curator_framework;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.api.transaction.CuratorOp;
+import org.apache.curator.framework.api.transaction.CuratorTransactionResult;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class CuratorFrameworkTest {
```

Comment:
This test can be executed in a native image without any metadata defined in the `metadata/org.apache.curator/curator-framework/5.5.0/reflect-config.json` file. It requires only zookeeper metadata.

Reply:
- The handling of the shaded package involves some classes that are not easily verified in unit tests, and it will take me some time to verify the remaining two classes.

- Normally Guava does not require additional GraalVM Reachability Metadata, but I remember I observed a failed nativeTest on GraalVM 22.3.1 and I needed to re-validate on GraalVM 23.1.0.  

- Additionally, for existing arbitrary test submodules, the GRM central repository's git is now unable to run nativeTest on GraalVM CE 23.1.0 For JDK21.

Reply:
- On https://github.com/linghengqian/graalvm-trace-metadata-smoketest/commit/41b06ca00b504b21f0d6d2a66ca803f8bf07b86c , I completed the further nativeTest on GraalVM CE 23.1.0 For JDK21, and neither the curator-framework nor the curator-client required additional GraalVM Reachability Metadata.